lib/bolt/cli.rb

```diff
@@ -652,8 +652,27 @@ def generate_types
     # Initializes a specified directory as a Bolt project and installs any modules
     # specified by the user, along with their dependencies
     def initialize_project
-      project    = Pathname.new(File.expand_path(options[:object] || Dir.pwd))
-      config     = project + 'bolt.yaml'
+      # Dir.pwd will return backslashes on Windows, but Pathname always uses
+      # forward slashes to concatenate paths. This results in paths like
+      # C:\User\Administrator/modules, which fail module install. This ensure
+      # forward slashes in the cwd path.
```

Comment:
Both `/` and `\` are valid path separators on Windows, and should not affect how the path is interpreted by any of the `File` methods, even if the two separators are used interchangeably. For example, the following shows that use of both path separators does not affect either `Pathname` or `File` methods:

```
path = Pathname.new('C:\Users\tom')
=> #<Pathname:C:\Users\tom>
bolt_path = path + 'bolt'
=> #<Pathname:C:\Users\tom/bolt>
bolt_path.exist?
=> true
File.exist?(bolt_path.to_s)
=> true
File.write(bolt_path + 'foo.txt', 'foo')
=> 3
```

Installing modules in the cwd with `bolt project init --modules puppetlabs/apache` also works on Windows with Bolt 2.21.0:

```
> New-BoltProject -Modules puppetlabs/apache
Successfully created Bolt project at C:\Users\tom/bolt
Successfully created Puppetfile at C:\Users\tom/bolt/Puppetfile
...
```